src/Containers/Microsoft.NET.Build.Containers/Registry.cs

```diff
@@ -447,32 +581,91 @@ private async Task<UriBuilder> StartUploadSessionAsync(string repository, string
             throw new ApplicationException(errorMessage);
         }
         cancellationToken.ThrowIfCancellationRequested();
-        return GetNextLocation(pushResponse);
+        var chunkSize = ParseRangeAmount(pushResponse) ?? ParseOCIChunkMinSizeAmount(pushResponse);
```

Comment:
We should start here. When starting an upload, registries can send `Range` or `OCI-Chunk-Min-Length` headers to tell us what chunk size they can handle. AWS for example sends a range of `0-10485760` (so 10MB) so we know what their upper-bound chunk size is. We do not however get a min-size from them.

Other registries also send one of these headers, but return invalid or useless values for them, so we have to validate that the values are > 0 in order to be used elsewhere. 

All of this data is now captured in a 'UploadInformation' structure that callers can mine for data for the next steps of the operation.